contrib/inventory/digital_ocean.py

```diff
@@ -441,16 +480,32 @@ def build_inventory(self):
 
         # add all droplets by id and name
         for droplet in self.data['droplets']:
-            for net in droplet['networks']['v4']:
-                if net['type'] == 'public':
-                    dest = net['ip_address']
-                else:
-                    continue
+            address = self.select_droplet_address(droplet)
+
+            do_id = droplet['id']
+            do_name = droplet['name']
+
+            if self.identifier == 'address':
+                dest = address
+            elif self.identifier == 'name':
+                dest = do_name
+                suffix = 0
+                while dest in self.inventory['all']['hosts']:
+                    suffix += 1
+                    dest = '%s-%d' % (do_name, suffix)
+            elif self.identifier == 'id':
+                dest = do_id
+            else:
+                raise ValueError('unknown identifier type %s configured' % self.identifier)
 
             self.inventory['all']['hosts'].append(dest)
 
-            self.inventory[droplet['id']] = [dest]
-            self.inventory[droplet['name']] = [dest]
+            if dest != do_id:
+                self.inventory[do_id] = [dest]
```

Comment:
@CFSworks didn't got this, why are we creating a group with the Droplet ID, IP or name? I think you made this to keep the old behaviour but I'm afraid ansible will discard hostvars, which are using **_dest_** since they don't match  **_do_id_** :/
I didn't tests this and maybe I'm confused.

Reply:
Yes I think you might confuse something here. This is just creating a group with one host.